launch/launcher.go

```diff
@@ -124,6 +146,86 @@ func (l *Launcher) env(process Process) error {
 	})
 }
 
+func (l *Launcher) execD(process Process) error {
+	execDs, err := l.buildpackFiles(process, ExecDDirName, supportsExecD)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find all exec.d executables in layers dir, '%s'", l.LayersDir)
+	}
+	for _, execD := range execDs {
+		if err := l.ExecD.ExecD(execD, l.Env); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func supportsExecD(bp Buildpack) bool {
+	if bp.API == "" {
+		return false
+	}
```

Comment:
I gather that if a buildpack implementing an earlier API writes an `exec.d`, it will be ignored. I am wondering why we don't give Windows buildpacks the same treatment i.e., can we check the runtime.GOOS here, and not fail if we find a Windows `exec.d`?

Reply:
I thought about that but decided to go this direction b/c buildpack spec 0.5 implies that `exec.d` should be available on windows. Therefore, if the lifecycle is not going to support exec.d on windows, I think it should be very explicit about it so buildpack authors aren't surprised to learn that they were silently omitted.

I would be happy to change the error to a warning instead if you think that is better.

Reply:
I see. In this case an error might be better, as a warning could be missed.